src/main/java/mcp/mobius/betterbarrels/client/render/TileEntityBaseRenderer.java

```diff
@@ -95,7 +95,7 @@ protected void renderStackOnBlock(ItemStack stack, ForgeDirection side, ForgeDir
 		GL11.glPushMatrix();
 
 		this.alignRendering(side, orientation, barrelPos);
-		this.moveRendering(size, posx, posy, 0);
+		this.moveRendering(size, posx, posy, -0.001);
```

Comment:
The problem with depth offset in the 1.7.10 old LWJGL rendering context is that the z-fighting re-appears at a distance from observer (player) because the relative scale of the offset diminishes with distance and the low precision of the z-buffer causes z-fight again.
To really prevent all possibility of z-fighting. The object shall be drawn on the very same vertices as the background and not on vertices with different coordinates. (like it is done for multi-layer textures, which don't z-fight each others because they share the very same vertices coordinates.)

Reply:
That had nothing to do with z-figting issue, it was fixed with polygon offset which respect non-linear depth in frustum. This depth offset is here because GT items for some reason render with an additional offset and are not visible on the barrel. Basically another dirty hack for GT.

Reply:
Ah fine then.